CMakeLists.txt

```
@@ -92,7 +92,7 @@ add_library(${PROJECT_NAME} SHARED
 )
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
 target_compile_options(${PROJECT_NAME} PRIVATE ${PROJECT_COMPILE_OPTIONS})
-ament_target_dependencies(${PROJECT_NAME}
+ament_target_dependencies(${PROJECT_NAME} SYSTEM
   ${THIS_PACKAGE_EXPORT_DEPENDS}
```

```suggestion
ament_target_dependencies(${PROJECT_NAME}
  ${THIS_PACKAGE_EXPORT_DEPENDS_ROS}
ament_target_dependencies(${PROJECT_NAME} SYSTEM
  ${THIS_PACKAGE_EXPORT_DEPENDS_EXTERNAL}
```

Maybe you can split them up this way with two different lists.  Make sure you update the ament_export_dependencies line to include both.